### ISSUE TITLE
Publish `semgrep` to an `apt` repository

### ISSUE THREAD

**Issue by nbrahms** [returntocorp/semgrep#372]
This is done when:
- [ ] `apt-get install semgrep` installs a runnable binary version of sgrep on:
  - [ ] Debian stretch (9.12)
  - [ ] Debian buster (10.3)
  - [ ] Debian jessie (8.11) ?? Or is this too old?
  - [ ] Ubuntu 16.04
  - [ ] Ubuntu 18.04
  - [ ] Ubuntu 19.10
- [ ] We have a documented build process for releasing new versions

**Comment by rcoh** [comment]
I'll start producing by a `deb` -- Unless someone has a friend in the debian committers it will be tricky to get it into mainline.

**Comment by aryx** [comment]
The main dependencies are Python3 (I think >3.6) for sgrep.py, and ocaml >=4.05 (and its ocamlrun bytecode interpreter) for the sgrep ocaml bytecode.
ocaml can also compile to native code in which case you can remove the runtime dependency to ocaml.
In theory you can also compile static binaries, but I'm not sure we need that for apt.

The Dockerfile at the top of the sgrep project should contain everything you need to build sgrep.
Let me know if you have any questions or are blocked on ocaml issues. For python issues you can ask to drew or isaac.

**Comment by brendongo** [comment]
Deprioritizing for now. For now an install script is sufficient.

**Comment by DrewDennison** [comment]
a `deb` is a great first step

**Comment by dlukeomalley** [comment]
Closing in favor of our current approach with `pip`.
